packages/contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -162,32 +162,40 @@ contract OptimismPortal is Initializable, ResourceMetering, Semver {
     }
 
     /**
-     * @notice Determine if an L2 Output is finalized.
+     * @notice Determine if an L2 Output is finalized for a given block number.
      *
      * @param _l2BlockNumber The number of the L2 block.
      */
-
     function isOutputFinalized(uint256 _l2BlockNumber) external view returns (bool) {
-        L2OutputOracle.OutputProposal memory proposal = L2_ORACLE.getL2Output(_l2BlockNumber);
-
-        if (proposal.outputRoot == bytes32(uint256(0))) {
+        L2OutputOracle.OutputProposal memory proposal;
```

Comment:
I don't think this try/catch stuff is really necessary. I would prefer that we keep this simple by doing the following:
1. Compute the block number that corresponds to the output that would've been published for a given L2 block number
2. Get the L2 output for that computed block number, it's fine if the error gets thrown if that output does not exist.
3. If it does exist, do the simple timestamp check

Reply:
Yeah, I was thinking about a similar alternative approach yesterday evening. Will do.